### ISSUE TITLE
Staging Review finding: Edit buttons don't have unique accessible names

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#104551]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** BAH-MBS-SelfServ
**Product:** Current - Pre-need Burial Requested change to Pre-Need Self Service
**Feature:** Pre-Need modernization and CaMEO Integration

## Findings details
**VA.gov Experience Standard - issue:** [User can identify an element and its state.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

There are two situations I encountered where "Edit" buttons don't have unique accessible names:

- When entering deceased people who have already been buried under this benefit (old list-and-loop pattern), each edit button has an identical `aria-label`.
- On the review page just before submitting, some edit buttons don't have any `aria-label` (just say "Edit") and some have an `aria-label` of "Edit [object Object]"

When a screen reader user moves through the page, they often encounter interactive elements like links and buttons out of context. Without that context and without visual cues to connect the to a certain section of the page, it's hard to know what a button is if it doesn't explicitly say what it's doing with unique text. (Example: I know I'm editing a person, but I've already entered three people. Which one am I editing?)

<details><summary>Link, screenshot or steps to recreate</summary>

Example:
![List-and-loop pattern for entering deceased persons' names. Each person has an Edit button.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/54528-Brian-3.png?raw=true)

</details>

## Recommended action

Ensure that each "Edit" button has a unique accessible name, either because there's only one edit button on the page or because each one has unique `aria-label` text. If you're not able to programmatically determine unique text ("Edit [name]") then an index is sufficient ("Edit person 1").

I'm guessing the "Edit [object Object]" one was a fun little JS bug --- might want to double check that page.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by MichelleDieudonne** [comment]
We found 2 bugs during our foundational and advanced accessibility testing prior to the staging review that address this item.  One will be worked next sprint and the other will be pulled in as capacity allows over the next few sprints. 
https://jira.devops.va.gov/browse/MBMS-87566
https://jira.devops.va.gov/browse/MBMS-87664

**Comment by MichelleDieudonne** [comment]
@shiragoodman Both of our bug fixes listed above are complete and we are closing this item.  Please let us know if you have any questions.
